### PR TITLE
Adds mdx to demo-gatsby

### PR DESCRIPTION
**Pull request by kendallstrautman**
Added a new `About` page in `demo-gatsby` to test the `gatsby-tinacms-mdx` package helpers. I started setting up the `useLocalMdx` hook, but wanted to just get this scaffolding in there so we can all test. I'm not super familiar with mdx so if there's a better way to render or work with it, feel free to update!

### CODE REVIEW

**Comment by kendallstrautman** [comment]
Just realized that Paul removed the `.d.ts` files in his PR, my bad. Can revert if needed.

File: packages/demo-gatsby/content/about.mdx [hunk 1]
```diff
@@ -0,0 +1,7 @@
+---
+title: About! whats up
+date: "2015-05-06T23:46:37.121Z"
+description: Test
+---
+
+Don't worry, be happy.
```
Comment (jsjoeio) [inline packages/demo-gatsby/content/about.mdx]: 😄

File: packages/demo-gatsby/src/pages/about.js [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+        label: "Body",
+        name: "rawMarkdownBody",
+        component: "markdown",
+      },
+    ],
+  }
+
+  const siteTitle = props.data.site.siteMetadata.title
+  const [data] = useLocalMdx(props.data.mdx, formOptions)
+  console.log("data", data)
```
Comment (jsjoeio) [inline packages/demo-gatsby/src/pages/about.js]:
```suggestion
```

File: packages/demo-gatsby/src/pages/about.js [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+
+*/
+import React from "react"
+import { graphql } from "gatsby"
+import { useLocalMdx } from "gatsby-tinacms-mdx"
```
Comment (DirtyF) [inline packages/demo-gatsby/src/pages/about.js]: based on https://github.com/tinacms/tinacms/pull/713#discussion_r375002577 shouldn'tit be `useLocalMdxForm` now?

File: packages/demo-gatsby/src/pages/about.js [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+      {
+        label: "Body",
+        name: "rawMarkdownBody",
+        component: "markdown",
+      },
+    ],
+  }
+
+  const siteTitle = props.data.site.siteMetadata.title
+  const [data] = useLocalMdx(props.data.mdx, formOptions)
```
Comment (DirtyF) [inline packages/demo-gatsby/src/pages/about.js]: Same here?

File: packages/demo-gatsby/src/pages/about.js (hunk 3 above)
Comment (kendallstrautman) [reply packages/demo-gatsby/src/pages/about.js]: yep! It's been fixed in the original PR I believe

**Pull request merged**